bitcart/errors.py

```diff
@@ -1,16 +1,42 @@
+import functools
+
+
 class BaseError(Exception):
     """Base error for all errors raised"""
 
-    pass
-
 
 class InvalidEventError(BaseError):
     """Daemon returned unsupported event"""
 
-    pass
-
 
 class LightningDisabledError(BaseError):
     """Lightning is disabled in daemon"""
 
-    pass
+
+class WebhookUnsupportedError(BaseError):
+    """Webhook support not installed"""
+
+
+class ConnectionFailedError(BaseError):
+    """Error connecting to the daemon"""
+
+
+class RequestError(BaseError):
+    """Base error for all errors returned from server"""
+
+
+class UnknownError(RequestError):
+    """Unknown error code returned from server"""
+
+
+@functools.lru_cache()
```

Comment:
default max_size is 128. Do we want to set it as None to max?

Reply:
From docs:

> If maxsize is set to None, the LRU feature is disabled and the cache can grow without bound.

I think LRU cache is useful here, as in many cases it will work like so:
Exception is generated and raised, and in exception handlers, users also catch that exception, so exception is returned from cache, as it is was used recently.
If we disable LRU, doesn't the cache grow indefinitely?
I don't think more than 128 exceptions limit will ever be reached anyway.